digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporter.scala

```diff
@@ -157,21 +213,96 @@ class TrafficSignCsvImporter extends CsvDataImporterOperations {
-  def createTrafficSigns(trafficSignAttributes: CsvAssetRow, roadLinks: Seq[VVHRoadlink]): Unit = {
-    val value = tryToInt(getPropertyValue(trafficSignAttributes, "value").toString)
-    val trafficSignType = getPropertyValue(trafficSignAttributes, "trafficSignType").toString.toInt
-    val bearing = tryToInt(getPropertyValue(trafficSignAttributes, "bearing").toString)
-    val trafficDirection = tryToInt(getPropertyValue(trafficSignAttributes, "trafficDirection").toString)
-    val twoSided = getPropertyValue(trafficSignAttributes, "twoSided").toString match { case "Kaksipuoleinen" => true case _ => false }
-    val lon = getPropertyValue(trafficSignAttributes, "lon").asInstanceOf[BigDecimal].toLong
-    val lat = getPropertyValue(trafficSignAttributes, "lat").asInstanceOf[BigDecimal].toLong
-    val additionalInfo = Some(getPropertyValue(trafficSignAttributes, "additionalInfo").toString)
-
-    trafficSignService.createFromCoordinates(lon, lat, TRTrafficSignType.apply(trafficSignType), value, Some(twoSided),
-      TrafficDirection.apply(trafficDirection), bearing, additionalInfo, roadLinks)
+
+  private def getPropertyValueOption(trafficSignAttributes: CsvAssetRow, propertyName: String) : Option[Any] = {
+    trafficSignAttributes.properties.find (prop => prop.columnName == propertyName).map(_.value)
+  }
+
+  private def generateBaseProperties(trafficSignAttributes: CsvAssetRow) : Set[SimpleTrafficSignProperty] = {
+    val valueProperty = tryToInt(getPropertyValue(trafficSignAttributes, "value").toString).map { value =>
+      SimpleTrafficSignProperty(valuePublicId, Seq(TextPropertyValue(value.toString)))}
+
+    val additionalInfo = getPropertyValue(trafficSignAttributes, "additionalInfo").toString
+    val additionalProperty = if(additionalInfo.nonEmpty)
+        Some(SimpleTrafficSignProperty(infoPublicId, Seq(TextPropertyValue(additionalInfo))))
+      else
+        None
+
+    val typeProperty = SimpleTrafficSignProperty(typePublicId, Seq(TextPropertyValue(TrafficSignType.applyTRValue(getPropertyValue(trafficSignAttributes, "trafficSignType").toString.toInt).OTHvalue.toString)))
+
+    Set(Some(typeProperty), valueProperty, additionalProperty).flatten
+  }
+
+  def createTrafficSigns(trafficSignAttributes: Seq[CsvAssetRowAndRoadLink]): Seq[AdditionalPanelInfo] = {
+
+    val signs = trafficSignAttributes.map { trafficSignAttribute =>
+      val properties = trafficSignAttribute.properties
+      val nearbyLinks = trafficSignAttribute.roadLink
+      val enrichedNearbyLinks = trafficSignAttribute.enrichedRoadLink
+      val optBearing = tryToInt(getPropertyValue(properties, "bearing").toString)
+      val twoSided = getPropertyValue(properties, "twoSided") match {
+        case "1" => true
+        case _ => false
+      }
+      val lon = getPropertyValue(properties, "lon").asInstanceOf[BigDecimal].toLong
+      val lat = getPropertyValue(properties, "lat").asInstanceOf[BigDecimal].toLong
+
+      if(enrichedNearbyLinks.nonEmpty) {
```

Comment:
With the possibility of sending also the decision boolean, we could refactor this code so it wouldn't have so much repeated code.